emulator/src/extensions/emu_args.c

```diff
@@ -30,57 +92,606 @@ emuArgs_t emulatorArgs = {
     .emulateTouch = false,
 };
 
-static const char doc[] = "Emulates a swadge";
+static const char mainDoc[] = "Emulates a swadge";
+
+// Long argument name definitions
+// These MUST be defined here, so that they are
+// the same in both options and argDocs
+static const char argFullscreen[] = "fullscreen";
+static const char argHideLeds[]   = "hide-leds";
+static const char argTouch[]      = "touch";
+static const char argHelp[]       = "help";
+static const char argUsage[]      = "usage";
 
 // clang-format off
-static const struct argp_option options[] =
+/**
+ * @brief The option definitions for getopt
+ */
+static const struct option options[] =
 {
-    { "fullscreen", 'f', 0, 0, "Open in fullscreen mode" },
-    { "hide-leds",  'l', 0, 0, "Don't draw simulated LEDs next to the display" },
-    { "touch",      't', 0, 0, "Simulate touch sensor readings with a virtual touchpad" },
+    { argFullscreen, no_argument, (int*)&emulatorArgs.fullscreen,   true },
+    { argHideLeds,   no_argument, (int*)&emulatorArgs.hideLeds,     true },
+    { argTouch,      no_argument, (int*)&emulatorArgs.emulateTouch, true },
+    { argHelp,       no_argument, NULL,                             'h'  },
+    { argUsage,      no_argument, NULL,                             0    },
+
     {0},
 };
-// clang-format on
 
-static struct argp argp = {options, parseOpt, NULL, doc};
+/**
+ * @brief Documentation strings for all options
+ */
+static const optDoc_t argDocs[] =
+{
+    {'f', argFullscreen, NULL, "Open in fullscreen mode" },
+    { 0,  argHideLeds,   NULL, "Don't draw simulated LEDs next to the display" },
+    {'t', argTouch,      NULL, "Simulate touch sensor readings with a virtual touchpad" },
+    {'h', argHelp,       NULL, "Give this help list" },
+    { 0,  argUsage,      NULL, "Give a short usage message" },
+};
+// clang-format on
 
 //==============================================================================
 // Functions
 //==============================================================================
 
-static error_t parseOpt(int key, char* arg, struct argp_state* state)
+/**
+ * @brief Handle a command-line option
+ *
+ * @param optName The name of the option
+ * @param arg The string value of this option's argument, or NULL
+ * @param optVal The value set in the struct, usually the short option
+ * @return true If the argument was handled successfully
+ * @return false If there was an error or the program should exit
+ */
+static bool handleArgument(const char* optName, const char* arg, int optVal)
 {
-    switch (key)
+    // Handle arguments with no short-option like this:
+    // if (optName == argUsage)
+    //{ doSomething(); return true }
+
+    // Handle options with a short-option here:
+    switch (optVal)
     {
-        case 'f':
+            // case 'x':
+            // doSomething();
+            // return true;
+
+        default:
+            return false;
+    }
+
+    return false;
+}
```

Comment:
This seems to do, nothing?

Reply:
Yeah, this is more of a placeholder since I don't have any short opts that need any extra handling. So far, they're all handled automatically by getopt setting a flag.

But, I do need to fix these returns because they should actually be true by default, and something else should have handled invalid args already.

Reply:
OK, returns are fixed now. And things that will use more args are on their way soon :eyes: